core/src/main/java/org/apache/iceberg/BaseFile.java

```diff
@@ -460,6 +460,12 @@ public ByteBuffer keyMetadata() {
 
   @Override
   public List<Long> splitOffsets() {
+    // If the last split offset is past the file size this means the split offsets are corrupted and
+    // should not be used
+    if (splitOffsets != null && splitOffsets[splitOffsets.length - 1] >= fileSizeInBytes) {
```

Comment:
maybe a length check also? not likely to be 0 but just in case

Reply:
Looks like `TestSplitPlanning` is the best place for verifying the behavior of split planning when the offsets are corrupted.